guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/RepositoryServiceImpl.java

```diff
@@ -98,74 +89,114 @@
     @Inject
     private BackwardCompatibleUtil backward;
 
-    private Map<String, Repository> configuredRepositories = new HashMap<String, Repository>();
-    private Map<Path, Repository> rootToRepo = new HashMap<Path, Repository>();
-
-    @SuppressWarnings("unchecked")
-    @PostConstruct
-    public void loadRepositories() {
-        final List<ConfigGroup> repoConfigs = configurationService.getConfiguration( REPOSITORY );
-        if ( !( repoConfigs == null || repoConfigs.isEmpty() ) ) {
-            for ( final ConfigGroup config : repoConfigs ) {
-                final Repository repository = repositoryFactory.newRepository( config );
-                configuredRepositories.put( repository.getAlias(), repository );
-                rootToRepo.put( repository.getRoot(), repository );
-                Collection<String> branches;
-                if ( repository instanceof GitRepository &&
-                        ( branches = repository.getBranches() ) != null &&
-                        !branches.isEmpty() ) {
-                    for ( String branch : branches ) {
-                        rootToRepo.put( repository.getBranchRoot( branch ), repository );
-                    }
+    @Inject
+    private ConfiguredRepositories configuredRepositories;
+
+    private Repository createRepository( final ConfigGroup repositoryConfig ) {
+        final Repository repository = repositoryFactory.newRepository( repositoryConfig );
+        configurationService.addConfiguration( repositoryConfig );
+        configuredRepositories.add( repository );
+        return repository;
+    }
+
+    public RepositoryInfo getRepositoryInfo( final String alias ) {
+        final Repository repo = getRepository( alias );
```

Comment:
This bit actually just moved and for some reason has been marked as new lines. You can find the original that is 1:1 in this file shown as deleted rows.
Based on a quick look I think the RepositoryInfo needs the OU name. We could cache the information like we cache Repositories.